drone_dji_sdkTypes.hpp

```diff
@@ -8,8 +8,49 @@
  * which case you do not need this file
  */
 
-namespace drone_dji_sdk {
+#include <base/Eigen.hpp>
+#include <base/Angle.hpp>
+#include <base/Time.hpp>
+
+namespace drone_dji_sdk
+{
+    enum COMMAND_ACTION
+    {
+        TAKEOFF_ACTIVATE,
+        PRE_LANDING_ACTIVATE,
+        LANDING_ACTIVATE,
+        GO_TO_ACTIVATE,
+        MISSION_ACTIVATE
+    };
+    struct VehicleSetpoint
+    {
+        base::Time time;
+        base::Vector3d position;
+        base::Angle heading;
+    };
+
+    struct Waypoint
```

Comment:
- in general, `reserved` means a list of fields that are currently unused. Don't show them in the public interface, and fill them with zeroes in the DJI data structure
- `index` is not needed here. You can infer the index when sending the waypoints to the SDK
- all angles are radians and use base::Angle. Convert to whatever DJI wants internally (e.g. yaw and gimbalPitch)
- define an Action struct and add a std::vector<Action> to this struct. Then do the conversion internally to what DJI wants.
- use an enum for turnMode

The convention is to use snake_case for fields in public structs used in orogen interfaces.